packages/jellyfish-testing/__tests__/icxorderbook.test.ts

```diff
@@ -0,0 +1,250 @@
+import { MasterNodeRegTestContainer } from '@defichain/testcontainers'
+import { Testing } from '@defichain/jellyfish-testing'
+import BigNumber from 'bignumber.js'
+import { icxorderbook } from '@defichain/jellyfish-api-core'
+
+const { ICXOrderStatus, ICXOrderType, ICXHTLCType } = icxorderbook
+
+const testing = Testing.create(new MasterNodeRegTestContainer())
+
+describe('ICX setup', () => {
+  beforeAll(async () => {
+    await testing.container.start()
+    await testing.container.waitForWalletCoinbaseMaturity()
+  })
+
+  afterAll(async () => {
+    await testing.container.stop()
+  })
+
+  it('should setup icx and init all public fields', async () => {
+    await testing.icxorderbook.setup()
+
+    expect(testing.icxorderbook.idDFI).toStrictEqual('0')
+    expect(testing.icxorderbook.idBTC).toStrictEqual('1')
+    expect(testing.icxorderbook.ICX_TAKERFEE_PER_BTC).toStrictEqual(new BigNumber(0.001))
+    expect(testing.icxorderbook.DEX_DFI_PER_BTC_RATE).toStrictEqual(new BigNumber(100))
+    expect(testing.icxorderbook.symbolDFI).toStrictEqual('DFI')
+    expect(testing.icxorderbook.symbolBTC).toStrictEqual('BTC')
+    expect(typeof testing.icxorderbook.accountDFI).toStrictEqual('string')
+    expect(testing.icxorderbook.accountDFI.length).toBeGreaterThan(0)
+    expect(typeof testing.icxorderbook.accountBTC).toStrictEqual('string')
+    expect(testing.icxorderbook.accountBTC.length).toBeGreaterThan(0)
+    expect(typeof testing.icxorderbook.poolOwner).toStrictEqual('string')
+    expect(testing.icxorderbook.poolOwner.length).toBeGreaterThan(0)
+  })
+})
+
+describe('ICX', () => {
+  beforeAll(async () => {
+    await testing.container.start()
+    await testing.container.waitForWalletCoinbaseMaturity()
+    await testing.icxorderbook.setup()
```

Comment:
this is okay for this test suite. But when refactoring ICX tests, it is better to list individual setup steps called here. It would give extra details about the setup without going into the icxorderbook.setup(). too much aggregation sometimes reduces context when it comes to testing. easy for the reader/reviewer.